core/src/main/java/org/infinispan/protostream/descriptors/EnumDescriptor.java

```diff
@@ -151,6 +153,53 @@ public String toString() {
       return "EnumDescriptor{fullName=" + getFullName() + '}';
    }
 
+   public ReservedNumbers getReservedNumbers() {
+      return reservedNumbers;
+   }
+
+   public Collection<String> getReservedNames() {
+      return reservedNames;
+   }
+
+   public void checkCompatibility(EnumDescriptor that, boolean strict, List<String> errors) {
+      for (EnumValueDescriptor thatValue : that.getValues()) {
+         if (reservedNumbers.get(thatValue.getNumber())) {
+            errors.add(Log.LOG.reservedNumber(thatValue.getNumber(), thatValue.getName(), that.getFullName()).getMessage());
+         }
+         if (reservedNames.contains(thatValue.getName())) {
+            errors.add(Log.LOG.reservedName(thatValue.getName(), that.getFullName()).getMessage());
+         }
+      }
+      for (EnumValueDescriptor thisValue : values) {
+         EnumValueDescriptor thatValue = that.valueByName.get(thisValue.getName());
+         if (thatValue == null) {
+            // Value was removed, make sure it has been reserved
+            if (!that.reservedNames.contains(thisValue.getName())) {
+               errors.add(Log.LOG.removedFieldNotReserved(thisValue.getFullName()));
+            }
+            if (!that.reservedNumbers.get(thisValue.getNumber())) {
+               errors.add(Log.LOG.removedFieldNotReserved(thisValue.getFullName(), thisValue.getNumber()));
+            }
+         } else {
+            if (thisValue.getNumber() != thatValue.getNumber()) {
+               errors.add(Log.LOG.modifiedFieldNumber(thisValue.getFullName(), thisValue.getNumber(), thatValue.getNumber()));
+            }
+         }
+      }
+      if (strict) {
```

Comment:
This is the same as in the `Descriptor` class, maybe we should extract this to a method in `AnnotatedDescriptorImpl`?

Reply:
Refactored